router/xgress_edge_tunnel_v2/tunneler.go

```diff
@@ -106,40 +112,18 @@ func (self *tunneler) Start(notifyClose <-chan struct{}) error {
 	self.serviceListener = intercept.NewServiceListener(self.interceptor, resolver)
 	self.serviceListener.HandleProviderReady(self.fabricProvider)
 
-	//go self.removeStaleConnections(notifyClose)
-
 	if err = self.env.GetRouterDataModel().SubscribeToIdentityChanges(self.env.GetRouterId().Token, self, true); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-//func (self *tunneler) removeStaleConnections(notifyClose <-chan struct{}) {
-//	ticker := time.NewTicker(time.Minute)
-//	defer ticker.Stop()
-//
-//	for {
-//		select {
-//		case <-ticker.C:
-//			var toRemove []string
-//			self.hostedServices.IterCb(func(key string, t *tunnelTerminator) {
-//				if t.closed.Load() {
-//					toRemove = append(toRemove, key)
-//				}
-//			})
-//
-//			for _, key := range toRemove {
-//				if t, found := self.hostedServices.Get(key); found {
-//					self.hostedServices.Remove(key)
-//					pfxlog.Logger().Debugf("removed closed tunnel terminator %v for service %v", key, t.context.ServiceName())
-//				}
-//			}
-//		case <-notifyClose:
-//			return
-//		}
-//	}
-//}
+func (self *tunneler) WaitForInitialized() {
+	for !self.initialized.Load() {
+		time.Sleep(100 * time.Millisecond)
```

Comment:
i feel like this should have a timeout associated -- shouldn't it?

Reply:
I can't think of a case where this wouldn't exit within a reasonable time frame, but added a timeout just to be safe.